src/state/store.jsx

```diff
@@ -18,20 +23,64 @@ const StateProvider = ({ children }) => {
           darkMode: currentState.darkMode,
           currentVideo: currentState.currentVideo,
           currentVideoProfile: currentState.currentVideoProfile,
+          loggedUser: currentState.loggedUser,
+          favorites: currentState.favorites,
         };
       case 'darkMode':
         return {
           videoList: currentState.videoList,
           darkMode: action.payload,
           currentVideo: currentState.currentVideo,
           currentVideoProfile: currentState.currentVideoProfile,
+          loggedUser: currentState.loggedUser,
+          favorites: currentState.favorites,
         };
       case 'play':
         return {
           videoList: currentState.videoList,
           darkMode: currentState.darkMode,
           currentVideo: action.video,
           currentVideoProfile: action.profile,
+          loggedUser: currentState.loggedUser,
+          favorites: currentState.favorites,
+        };
+      case 'login':
+        return {
+          videoList: currentState.videoList,
+          darkMode: currentState.darkMode,
+          currentVideo: currentState.currentVideo,
+          currentVideoProfile: currentState.currentVideoProfile,
+          loggedUser: action.payload,
+          favorites: currentState.favorites,
+        };
+      case 'logout':
```

Comment:
Do you need to still be saving all the data or you could empty the whole structures related to videos?